### ISSUE TITLE
Enhancement Request: Add extra Difficulty input argument

### ISSUE THREAD

**Issue by Mandelhoff** [zack-bitcoin/amoveo-c-miner#5]
Enhancement Request: Add extra Difficulty input argument

Two difficulties sent to the miner:
1) Network diff - used in the sha256(bhash + diff1 + nonce)
2) Pool share diff - used to prove to a pool that work is done even if this is not "block quality" results.

A second input argument for difficulty is needed for a "share" based mining pool. Your prototype Amoveo mining pool will just send the difficulty twice as it only accepts block quality results.

**Comment by zack-bitcoin** [comment]
Yes. As the difficulty goes up, a feature like this becomes more important.
